ER-72-API-SignIn.

amansiid opened the pull request:
Finish API Sign In requirements: 
Endpoint URL => /api/authentication/sign-in

amansiid committed:
Finish API Sign In Requirement.

leduyantran commented on src_code/src/controllers/interfaces/IAuthentication.js:
```diff
@@ -0,0 +1,5 @@
+
+module.exports = {
+    SignIn: signInService.SignIn
+}
```
Let's add the newline here at the end of the file 👍

leduyantran commented on src_code/src/controllers/services/Authentication/SignIn.js:
```diff
@@ -0,0 +1,82 @@
+
+module.exports = {
+    SignIn,
+}
```
Let's add the newline here at the end of the file 👍Good explanations and steps through!

leduyantran commented on src_code/src/utils/interfaces/IDBHelperFunctions.js:
```diff
@@ -0,0 +1,7 @@
+/** Exports the functions */
+module.exports = {
+    GetUserInfoBasedOnEmailAddress: dbHelperFunctions.GetUserInfoBasedOnEmailAddress,
+}
```
Let's add the newline here at the end of the file 👍

leduyantran commented on src_code/src/utils/interfaces/IResponseBuilder.js:
```diff
@@ -0,0 +1,15 @@
+    NotFound: responseBuilder.NotFound,
+    ServerError: responseBuilder.ServerError,
+    BadRequest: responseBuilder.BadRequest,
+}
```
Let's add the newline here at the end of the file 👍

leduyantran commented on src_code/src/utils/services/helper_functions/db_helper_functions/DBHelperFunctions.js:
```diff
@@ -0,0 +1,47 @@
+
+module.exports ={
+    GetUserInfoBasedOnEmailAddress,
+}
```
Let's add the newline here at the end of the file 👍

leduyantran commented on src_code/src/utils/services/helper_functions/db_helper_functions/DBHelperFunctions.js:
```diff
@@ -0,0 +1,47 @@
+        /** Initailize an user object query from database */
+        const user = await db.select(
+            "PK_USER_ID",
+            "FK_CDE_ROLE", 
```
As we discussed in the database schema, we will not using the user_role table anymore, so instead of "FK_CDE_ROLE", we can just named it "USER_ROLE" instead.

leduyantran commented on src_code/src/utils/services/helper_functions/db_helper_functions/DBHelperFunctions.js:
```diff
@@ -0,0 +1,47 @@
+        }
+
+        /** Get user role description */
+        const userRole = await db.select("DESC_ROLE").from("user_role").where("PK_CDE_ROLE", "=", user.FK_CDE_ROLE).first();  
```
Same as above, as we discussed in the database schema, we will not using the user_role table anymore, so we won't need to get the code role from the user_role table. You can remove this line.

leduyantran commented on src_code/src/utils/services/helper_functions/db_helper_functions/DBHelperFunctions.js:
```diff
@@ -0,0 +1,47 @@
+        /** If there is user, then we return there information */
+        const responseObject = {
+            userId: user.PK_USER_ID,
+            userRole: userRole.DESC_ROLE,
```
This line here can just be userRole: user.USER_ROLE,

leduyantran committed:
Resolve comments and QA Validation and response.

Changed FK_CDE_ROLE to USER_ROLE and remove get userRole from user_role table code because we no longer need user_role table.

leduyantran commented:
All the comments are addressed!

leduyantran reviewed (approved):
Good job!

mohammedking11-spu commented:
It is reasonable to follow along with Le's comments, I took a look at them and they make sense. Great job so far.

mohammedking11-spu reviewed (approved):
Overall, very nicely written, even better when Le committed some changes. Good practice of spacing as well as when it comes to defining the variables. Good job on sticking and abiding with the code practices rules we established. Good comments, overall, good job!
